plugins/outputs/zabbix/zabbix_test.go

```diff
@@ -469,36 +472,59 @@ func TestZabbix(t *testing.T) {
 			}
 			require.NoError(t, z.Init())
 
+			outerErrs := make(chan error, 1)
 			wg := sync.WaitGroup{}
 			wg.Add(1)
 			go func() {
+				defer wg.Done()
+
 				success := make(chan zabbixRequest, 1)
+				innerErrs := make(chan error, 1)
 
 				go func() {
-					success <- listenForZabbixMetric(t, listener, len(test.zabbixMetrics) == 0)
+					req, err := listenForZabbixMetric(t, listener, len(test.zabbixMetrics) == 0)
+					if err != nil {
+						innerErrs <- err
+					} else {
+						success <- req
+					}
```

Comment:
How about encapsulating the `listener` stuff in a `zabbixMockServer` struct and record the received messages and errors there instead of utilizing channels to pass around data. We then perform the test, wait for the required data (with timeout) and then check the recorded internal data (and errors). This also allows to stick with `require` instead of reintroducing `assert`... What do you think?

Reply:
Utilized `zabbixMockServer` and simplfied things.